### PR TITLE
Don't reuse passmanagers across compilations

### PR DESCRIPTION
**Pull request by gbaraldi**
According to https://reviews.llvm.org/D137149, it's not legal to reuse the pipeline across compilations.
While we want to get back into doing that, from some very quick measurements I wasn't able to measure any difference (It wasn't extensive) 
In order to stop doing this, I will propose some discussions upstream in order to make passes aware that their state should go in the AnalysisManager.

This also reinstates the StandardInstrumentation with LLVM16 built

### CODE REVIEW

File: src/pipeline.cpp [hunk 1]
```diff
@@ -743,12 +743,13 @@ PIC.addClassToPassName(decltype(CREATE_PASS)::name(), NAME);
...
-NewPM::NewPM(std::unique_ptr<TargetMachine> TM, OptimizationLevel O, OptimizationOptions options) :
+NewPM::NewPM(std::unique_ptr<TargetMachine> TM, OptimizationLevel O, LLVMContext &ctx, OptimizationOptions options) :
     TM(std::move(TM)),
 #if JL_LLVM_VERSION < 160000
     SI(false),
     PIC(createPIC(SI)),
 #else
+    SI(ctx, false),
     PIC(createPIC()),
 #endif
```
Comment (vchuravy) [inline src/pipeline.cpp:754]:
```suggestion
#if JL_LLVM_VERSION < 160000
    SI(false),
#else
    SI(ctx, false),
#endif
    PIC(createPIC(SI)),
```

I think now PIC can take SI again?

Comment (gbaraldi) [reply src/pipeline.cpp:754]: Oh, forgot to fix the thing that made me do the whole PR :)

File: src/pipeline.cpp [hunk 2]
```diff
@@ -743,14 +735,14 @@ PIC.addClassToPassName(decltype(CREATE_PASS)::name(), NAME);
...
 
-NewPM::NewPM(std::unique_ptr<TargetMachine> TM, OptimizationLevel O, OptimizationOptions options) :
+NewPM::NewPM(std::unique_ptr<TargetMachine> TM, OptimizationLevel O, LLVMContext &ctx, OptimizationOptions options) :
     TM(std::move(TM)),
 #if JL_LLVM_VERSION < 160000
     SI(false),
-    PIC(createPIC(SI)),
 #else
-    PIC(createPIC()),
+    SI(ctx, false),
```
Comment (pchintalapudi) [inline src/pipeline.cpp:743]: While you're at it might as well refactor to take the FAM as well, for the cfg verification.

**Review by vtjnash — APPROVED** [review]
If we don't need to reuse things, that it is certainly easier not to

**Comment by gbaraldi** [comment]
The annoying part is that we pay the cost of rebuilding those data structures every run. But it doesn't seem to be affecting things too much

**Pull request merged**